src/components/EditPost.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <div class="container">
+    <div class="row">
+      <div class="col-12 text-center">
+        <h2 class="pt-3">Update Post</h2>
+      </div>
+    </div>
+    <div class="row">
+      <div class="col-3"></div>
+      <div class="col-6">
+        <form @submit.prevent="updatePost">
+          <div class="form-group">
+            <label>ImageURL:</label>
+            <input
+              type="text"
+              v-model="post.image"
+              class="form-control"
+              required
+            />
+          </div>
+          <div class="form-group">
+            <label for="tags">Tags</label>
+            <multiselect
+              v-model="post.selectedTags"
+              :options="allTags"
+              :multiple="true"
+              :close-on-select="true"
+              placeholder="Select tags"
+              class="form-control"
+              required
+            ></multiselect>
+          </div>
+
+          <div class="form-group">
+            <label>Description:</label>
+            <textarea
+              v-model="post.description"
+              class="form-control"
+              required
+            ></textarea>
+            <button type="submit" class="btn btn-primary mt-3">Update</button>
+          </div>
+        </form>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Multiselect from "vue-multiselect";
+import axios from "axios";
+
+export default {
+  name: "EditPost",
+  components: {
+    Multiselect,
+  },
+  data() {
+    return {
+      post: {
+        id: "",
+        image: "",
+        selectedTags: [],
+        description: "",
+        created_at: "", 
+      },
+      allTags: [
+        "transport",
+        "sports",
+        "people",
+        "nightlife",
+        "nature",
+        "food",
+        "fashion",
+        "city",
+        "cats",
+        "business",
+        "other",
+      ], // Store all available tags
+    };
+  },
+  created() {
+    this.fetchPost();
+    this.fetchAllTags();
+  },
+  methods: {
+    fetchPost() {
```

Comment:
This function should be stored in the `store` as an action. It should take the `postId` as an argument.
You should put all functions which perform axios requests in the action section of the store.